torchvision/csrc/io/image/cpu/decode_gif.cpp

```diff
@@ -121,30 +108,65 @@ torch::Tensor decode_gif(const torch::Tensor& encoded_data) {
   auto out = torch::empty(
       {int64_t(num_images), 3, int64_t(out_h), int64_t(out_w)}, options);
   auto out_a = out.accessor<uint8_t, 4>();
-
   for (int i = 0; i < num_images; i++) {
     const SavedImage& img = gifFile->SavedImages[i];
-    const GifImageDesc& desc = img.ImageDesc;
-    TORCH_CHECK(
-        desc.Width == out_w && desc.Height == out_h,
-        "All images in the gif should have the same dimensions.");
 
+    GraphicsControlBlock gcb;
+    DGifSavedExtensionToGCB(gifFile, i, &gcb);
+
+    const GifImageDesc& desc = img.ImageDesc;
     const ColorMapObject* cmap =
         desc.ColorMap ? desc.ColorMap : gifFile->SColorMap;
     TORCH_CHECK(
         cmap != nullptr,
         "Global and local color maps are missing. This should never happen!");
 
+    // When going from one image to another, there is a "disposal method" which
+    // specifies how to handle the transition. E.g. DISPOSE_DO_NOT means that
+    // the current image should essentially be drawn on top of the previous
+    // canvas. The pixels of that previous canvas will appear on the new one if
+    // either:
+    // - a pixel is transparent in the current image
+    // - the current image is smaller than the canvas, hence exposing its pixels
+    // The "background" disposal method means that the current canvas should be
+    // set to the background color.
+    // We only support these 2 modes and default to "background" when the
+    // disposal method is unspecified, or when it's set to "DISPOSE_PREVIOUS"
+    // which according to GIFLIB is not widely supported.
+    // (https://giflib.sourceforge.net/whatsinagif/animation_and_transparency.html).
+    if (i > 0 && gcb.DisposalMode == DISPOSE_DO_NOT) {
+      for (int h = 0; h < gifFile->SHeight; h++) {
```

Comment:
Here the copy is done in ranges (0, gifFile->SHeight) and (0, gifFile->SWidth)?
I mean if do we copy the full spatial size of data or top-left part of it.
In any case we may probably use torch API to copy data here?
```c++
out[i] = out[i - 1];
// or this but have to check left-hand size
out.index({i, "...", at::indexing::Slice(0, gifFile->SHeight), at::indexing::Slice(0, gifFile->SWidth)}) = out.index({i - 1, "...", at::indexing::Slice(0, gifFile->SHeight), at::indexing::Slice(0, gifFile->SWidth)});
```